src/proxy_handler.cpp

```diff
@@ -0,0 +1,100 @@
+#include <string>
+#include <istream>
+#include <ostream>
+#include <boost/asio.hpp>
+#include "proxy_handler.h"
+
+using boost::asio::ip::tcp;
+RequestHandler::Status ProxyHandler::Init(const std::string& uri_prefix,
+         const NginxConfig& config) {
+	uri_prefix_ = uri_prefix;
+	RequestHandler::ParseConfig(config);
+	host = ProxyHandler::getHost();
+	portno = ProxyHandler::getPort();
```

Comment:
Like I mentioned above, just move the code in these functions directly into the constructor since you don't need them anywhere else.